### PR TITLE
feat(Tinkercad): add presence

### PR DESCRIPTION
**Pull request by ION606**
## Description 
Tinkercad is a free, easy-to-use app for 3D design, electronics, and coding

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

image

</details>

from https://github.com/PreMiD/Presences/issues/7698

### CODE REVIEW

File: websites/T/Tinkercad/handleThing.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// import { Assets } from "./presence";
```
Comment (theusaf) [inline websites/T/Tinkercad/handleThing.ts:1]:
```suggestion
```

File: websites/T/Tinkercad/handleThing.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+export function handleThing(
+	presenceData: PresenceData,
+	link: string,
+	isEdit: boolean
+): void {
+	if (!link) {
+		presenceData.state = "Browing the Community Gallery";
+		presenceData.buttons = [{ label: "View", url: document.location.href }];
```
Comment (theusaf) [inline websites/T/Tinkercad/handleThing.ts:10]: Make these buttons more specific. What do these buttons lead the user to view?

File: websites/T/Tinkercad/dashboard.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+			presenceData.details = "viewing codeblocks";
+			presenceData.smallImageKey = Assets.CodeblockIco;
+			presenceData.smallImageText = "code";
+			break;
+		}
+		case "bin": {
+			presenceData.details = "viewing Trash Can";
+			presenceData.smallImageKey = Assets.TrashCanIco;
+			presenceData.smallImageText = "trash";
+			break;
```
Comment (theusaf) [inline websites/T/Tinkercad/dashboard.ts:40]: These details should be capitalized to be consistent with the rest of the presence

File: websites/T/Tinkercad/presence.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import { handleCodeblocks, handleThing } from "./handleThing";
+
+const presence = new Presence({
+	clientId: "1217153856665026580",
+});
+
+export const enum Assets {
+	Logo = "https://i.imgur.com/XXez3yK.png",
+	ThreeDIco = "https://icon-library.com/images/3d-icon-png/3d-icon-png-17.jpg",
+	CircuitIco = "https://img.freepik.com/premium-vector/circuit-board-linecpuchip-icon-logo-illustration-vector-design_598213-3208.jpg",
```
Comment (theusaf) [inline websites/T/Tinkercad/presence.ts:11]: This asset should be a square (512x512)

File: websites/T/Tinkercad/presence.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	CodeblockIco = "https://cdn.iconscout.com/icon/free/png-512/code-280-460136.png",
+	TrashCanIco = "https://www.nicepng.com/png/detail/208-2086588_trash-can-icon.png",
```
Comment (theusaf) [inline websites/T/Tinkercad/presence.ts:13]: Ditto

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/T/Tinkercad/handleThing.ts (hunk 2 above)
Comment (ION606) [reply websites/T/Tinkercad/handleThing.ts:9]: the gallery?

Comment (ION606) [reply websites/T/Tinkercad/handleThing.ts:9]: actually, I'm gonna remove it

Comment (theusaf) [reply websites/T/Tinkercad/handleThing.ts:9]: This comment also applies for the other buttons in this presence

File: websites/T/Tinkercad/metadata.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+	"service": "Tinkercad",
+	"description": {
+		"en": "Tinkercad is a free, easy-to-use app for 3D design, electronics, and coding"
+	},
+	"url": "www.tinkercad.com",
+	"matches": [
+		"*://www.tinkercad.com/*"
+	],
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/XXez3yK.png",
```
Comment (theusaf) [inline websites/T/Tinkercad/metadata.json:16]: Image appears to be removed, or a temporary issue by Imgur.
```suggestion
	"logo": "https://i.imgur.com/2Nyp2DT.png",
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**